chroma_core/services/job_scheduler/job_scheduler.py

```diff
@@ -1745,3 +1755,103 @@ def update_lnet_configuration(self, lnet_configuration_list):
     @property
     def CommandPlan(self):
         return CommandPlan(self._lock_cache, self._job_collection)
+
+    def configure_stratagem(self, stratagem_data):
+        with self._lock:
+            configuration_data = {
+                "state": "unconfigured",
+                "filesystem_id": stratagem_data.get("filesystem"),
+                "interval": stratagem_data.get("interval"),
+                "report_duration": stratagem_data.get("report_duration"),
+                "purge_duration": stratagem_data.get("purge_duration"),
+            }
+
+            # The filesystem_id may come in as the fs name or the fs id. In terms of storing information in the database, the fs id should always be used.
+            fs_identifier = str(configuration_data.get("filesystem_id"))
+            fs_id = get_fs_id_from_identifier(fs_identifier)
+
+            if not fs_id:
+                raise Exception("No matching filesystem for {}".format(fs_identifier))
+
+            configuration_data["filesystem_id"] = fs_id
+
+            matches = StratagemConfiguration.objects.filter(filesystem_id=fs_id)
```

Comment:
I think this would be clearer as

```python
try:
    stratagem_configuration = StratagemConfiguration.objects.get(filesystem_id=fs_id)
    stratagem_configuration.update(**configuration_data)
except StratagemConfiguration.DoesNotExist:
    stratagem_configuration = StratagemConfiguration.objects.create(**configuration_data)
```

Reply:
In the try, shouldn't we update and then pull the data again to have the new object before adding it to cache?

Reply:
Does that make a difference? I would have expected .save() to be called, but didn't see it happening before.

Reply:
Maybe something like:
```
try:
    stratagem_configuration = StratagemConfiguration.objects.get(filesystem_id=fs_id)
    stratagem_configuration.update(**configuration_data)
    stratagem_configuration = StratagemConfiguration.objects.get(filesystem_id=fs_id)
except StratagemConfiguration.DoesNotExist:
    stratagem_configuration = StratagemConfiguration.objects.create(**configuration_data)
```